### PR TITLE
spec: clarify URLs type, templates usage and mandatory {ext} in discovery process.

### PR DESCRIPTION
**Pull request by sgotti**

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -224,21 +224,34 @@ Furthermore, attributes other than the name may be required to unambiguously ide
...
+    https://{name}-{version}-{os}-{arch}.{ext}
+
+First, try to fetch the app container image by rendering the above template (with `{ext}` rendered to `aci`) and directly retrieving the resulting URL.
 
 For example, given the app name `example.com/reduce-worker`, with version `1.0.0`, arch `amd64`, and os `linux`, try to retrieve:
 
     https://example.com/reduce-worker-1.0.0-linux-amd64.aci
 
 If this fails, move on to meta discovery.
-If this succeeds, try fetching the signature using the same template but with a `.sig` extension:
```
Comment (philips) [inline SPEC.md]: hrm, the one think I realized is that .sig is sort of ambiguous. Perhaps it should be .aci.sig?

Comment (sgotti) [inline SPEC.md]: @philips That makes sense.

Comment (philips) [inline SPEC.md]: we ended up changing this to aci.asc

**Comment by philips** [comment]
These are good changes, I am rebasing them now.

**Comment by philips** [comment]
Replaced with #220

**Pull request closed without merging**